### PR TITLE
Allow previewing background adjustments at player loading screen

### PR DESCRIPTION
**Pull request by UselessToucan**
I extracted some common stuff from Player/PlayerLoader to new PlayerBase class and changed PlayerLoader's background behavior. [Here is how it works now.](https://youtu.be/xAwEADi6gjA)

- [x] Depends on #2073

### CODE REVIEW

**Comment by peppy** [comment]
You've caused the loading of everything twice and removed async loading. We cannot do either of these things.

**Comment by UselessToucan** [comment]
I checked `PlayerBase`, `PlayerLoader`, `Player` code and I do not understand what do you mean. Please explain it in more details.

> loading of everything twice

If you are about storyboard, then the only call which actually loads something is [this one](https://github.com/ppy/osu/pull/2045/files#diff-500a255129fba9ee4be35ab9356654e6R188). Note that it was not async before.


> removed async loading

Well, I just don't understand which async loading do you mean. [The player loading is async](https://github.com/UselessToucan/osu/blob/PlayerBase/osu.Game/Screens/Play/PlayerLoader.cs#L56), storyboard loading isn't, but you can read about it above.

**Comment by peppy** [comment]
Everything inside that `load` method was previous async loaded while `PlayerLoader` is visible. You moved a portion of this to a base class which is loaded twice.

The storyboard is being loaded in the `UpdateBackgroundElements` call, which means it can potentially be loaded once in `PlayerLoader`  and again when `Player` loads. Also you are loading samples in `PlayerLoader` which are not used until Player, moving more of the load process to `PlayerLoad`.

**Comment by UselessToucan** [comment]
> Everything inside that load method was previous async loaded while PlayerLoader is visible. You moved a portion of this to a base class which is loaded twice.

I need config values on two screens(`Player`, `PlayerLoader`). What is wrong with loading things wherever I need them?


> The storyboard is being loaded in the UpdateBackgroundElements call, which means it can potentially be loaded once in PlayerLoader

Since `PlayerLoader.StoryboardContainer` is always null, [this condition](https://github.com/ppy/osu/pull/2045/files#diff-044e6034fca0f93820638bdad9a5255bR84) is always true for `PlayerLoader`.


> Also you are loading samples in PlayerLoader which are not used until Player

Yes, this needs to be moved to `Player`.

**Comment by peppy** [comment]
Even if that's the case for storyboards, the logic is so convoluted I couldn't see that.

We definitely need some restructuring here. Hopefully moving all storyboard loading to `Player` itself (make `UpdateBackgroundElements` virtual?)

Also I'd recommend renaming `PlayerBase` to `ScreenWithBeatmapBackground` or similar.

**Comment by peppy** [comment]
Adding dependency on #2073.

File: osu.Game/Screens/Play/ScreenWithBeatmapBackground.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            DimLevel = config.GetBindable<double>(OsuSetting.DimLevel);
+            BlurLevel = config.GetBindable<double>(OsuSetting.BlurLevel);
+            ShowStoryboard = config.GetBindable<bool>(OsuSetting.ShowStoryboard);
+
+            MouseWheelDisabled = config.GetBindable<bool>(OsuSetting.MouseDisableWheel);
+
+            UserAudioOffset = config.GetBindable<double>(OsuSetting.AudioOffset);
+        }
+
+        protected void ConfigureBackgroundUpdate()
```
Comment (peppy) [inline osu.Game/Screens/Play/ScreenWithBeatmapBackground.cs]: why is this a thing? it's called from the same place in both consumers of this screen. is there something i'm missing or can this just be moved to `OnEntering`?

Comment (UselessToucan) [reply osu.Game/Screens/Play/ScreenWithBeatmapBackground.cs]: You are correct. Can be moved to `ScreenWithBeatmapBackground.OnEntering`.

**Comment by peppy** [comment]
Made a minor change to avoid spilling unnecesssary bindables into the new class, but looks good otherwise 👍

**Review by peppy — APPROVED** [review]

**Pull request merged**